port/cpl_vsil_curl.cpp

```diff
@@ -1374,7 +1373,17 @@ vsi_l_offset VSICurlHandle::GetFileSizeOrHeaders(bool bSetError,
                         CPLParseNameValue(papszHeaders[i], &pszKey);
                     if (pszKey && pszValue)
                     {
-                        m_aosHeaders.SetNameValue(pszKey, pszValue);
+                        if (bGetHeaders)
+                        {
+                            m_aosHeaders.SetNameValue(pszKey, pszValue);
+                        }
+                        if (EQUAL(pszKey, "Cache-Control") &&
+                            EQUAL(pszValue, "no-cache") &&
+                            CPLTestBool(CPLGetConfigOption(
+                                "CPL_VSIL_CURL_HONOR_CACHE_CONTROL", "YES")))
+                        {
+                            m_bCached = false;
```

Comment:
@rouault does this mean that the results are not cached at all? Or that cached results aren't used without first validating with the server?

Reply:
It checks if the server returns a 'Cache-Control: no-cache' response header. If it does, then, *when the file is closed*, any cached data about that file is invalidated. Which is the behaviour you could already get with the [CPL_VSIL_CURL_NON_CACHED](https://gdal.org/en/latest/user/configoptions.html#config-CPL_VSIL_CURL_NON_CACHED) config option.